File: wc-tool/python/ccwc.py
import sys
import argparse


def count_bytes(file_content):
	byte_count = len(file_content)
	return byte_count


def count_lines(file_content):
	line_count = file_content.count(b'\n')
	return line_count


def count_words(file_content):
	words = file_content.decode('utf-8').split()
	words_count = len(words)
	return words_count


def count_characters(file_content):
	char_count = len(file_content.decode('utf-8'))
	return char_count


def main():
	parser = argparse.ArgumentParser(
		prog='Custom Word Count Tool by @abidmuin',
		description='Print bytes, lines, words and characters counts for each FILE',
		epilog='Author: https://github.com/abidmuin',
		formatter_class=argparse.RawDescriptionHelpFormatter,
	)
	parser.add_argument('-c', '--bytes', action='store_true', help='print the byte counts')
	parser.add_argument('-l', '--lines', action='store_true', help='print the newline counts')
	parser.add_argument('-w', '--words', action='store_true', help='print the word counts')
	parser.add_argument('-m', '--chars', action='store_true', help='print the character counts')
	parser.add_argument('file', metavar='file', type=str, nargs='?', help='File path')
	args = parser.parse_args()

	if args.file:
		with open(args.file, 'rb') as file:
			file_content = file.read()
	elif not sys.stdin.isatty():
		file_content = sys.stdin.buffer.read()
	else:
		print("No standard input found.")
		sys.exit(1)

	if args.bytes:
		result = count_bytes(file_content)
		if args.file:
			print(f'{result} {args.file}')
		else:
			print(result)
	elif args.lines:
		result = count_lines(file_content)
		if args.file:
			print(f'{result} {args.file}')
		else:
			print(result)
	elif args.words:
		result = count_words(file_content)
		if args.file:
			print(f'{result} {args.file}')
		else:
			print(result)
	elif args.chars:
		result = count_characters(file_content)
		if args.file:
			print(f'{result} {args.file}')
		else:
			print(result)
	else:
		bytes_count = count_bytes(file_content)
		lines_count = count_lines(file_content)
		words_count = count_words(file_content)
		if args.file:
			print(f'{bytes_count} {lines_count} {words_count} {args.file}')
		else:
			print(f'{bytes_count} {lines_count} {words_count}')


if __name__ == '__main__':
	main()

File: wc-tool/java/ccwc.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.charset.Charset;

public class ccwc {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("usage: Custom Word Count Tool by @abidmuin [-h] [-c] [-l] [-w] [-m] [file]");
            System.exit(1);
        }

        String option = args[0];
        String filename = (args.length > 1) ? args[1] : null;

        try {
            if (filename != null) {
                processFile(option, filename);
            } else {
                processStdin(option);
            }
        } catch (IOException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }

    private static void processFile(String option, String filename) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(filename, Charset.defaultCharset()))) {
            switch (option) {
                case "-c":
                    countBytes(reader, filename);
                    break;
                case "-l":
                    countLines(reader, filename);
                    break;
                case "-w":
                    countWords(reader, filename);
                    break;
                case "-m":
                    countCharacters(reader, filename);
                    break;
                default:
                    printDefaultStats(reader, filename);
                    break;
            }
        }
    }

    private static void processStdin(String option) throws IOException {
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
            switch (option) {
                case "-c":
                    countBytes(reader, null);
                    break;
                case "-l":
                    countLines(reader, null);
                    break;
                case "-w":
                    countWords(reader, null);
                    break;
                case "-m":
                    countCharacters(reader, null);
                    break;
                default:
                    printDefaultStats(reader, null);
                    break;
            }
        }
    }

    private static void countBytes(BufferedReader reader, String filename) throws IOException {
        int count = 0;
        int value;
        while ((value = reader.read()) != -1) {
            count++;
        }
        printResult(count, filename);
    }

    private static void countLines(BufferedReader reader, String filename) throws IOException {
        int count = 0;
        while (reader.readLine() != null) {
            count++;
        }
        printResult(count, filename);
    }

    private static void countWords(BufferedReader reader, String filename) throws IOException {
        int count = 0;
        String line;
        while ((line = reader.readLine()) != null) {
            String[] words = line.split("\\s+");
            count += words.length;
        }
        printResult(count, filename);
    }

    private static void countCharacters(BufferedReader reader, String filename) throws IOException {
        int count = 0;
        int value;
        while ((value = reader.read()) != -1) {
            count++;
        }
        printResult(count, filename);
    }

    private static void printDefaultStats(BufferedReader reader, String filename) throws IOException {
        int lineCount = 0;
        int wordCount = 0;
        int byteCount = 0;
        String line;
        while ((line = reader.readLine()) != null) {
            lineCount++;
            String[] words = line.split("\\s+");
            wordCount += words.length;
            byteCount += line.length();
        }
        printResult(lineCount, wordCount, byteCount, filename);
    }

    private static void printResult(int count, String filename) {
        if (filename != null) {
            System.out.printf("%8d %s%n", count, filename);
        } else {
            System.out.printf("%8d%n", count);
        }
    }

    private static void printResult(int lines, int words, int bytes, String filename) {
        if (filename != null) {
            System.out.printf("%8d %8d %8d %s%n", lines, words, bytes, filename);
        } else {
            System.out.printf("%8d %8d %8d%n", lines, words, bytes);
        }
    }
}